docs/best-practices/folders-structure.md

```diff
@@ -80,3 +90,39 @@ The main difference between a KMM and a native project is that the presentation
             │   └── model
             └── FeatureProvider.kt # DI Component. Provides interactor mostly
 ```
+</TabItem>
+
+<TabItem value="angular">
+
+### Angular Project structure
+
+```txt
+.
+└── project
+    └── src
+        ├── app                      # Angular stuff
+        │   ├── feature-name         # UI module
+        │   │   ├── component-name   # Includes component + html + css + view-state + test
+        │   │   ├── feature.module.ts
+        │   │   └── feature-routing.module.ts
+        │   ├── shared               # A folder to place directives, pipes, shared components and every cross-feature class
+        │   ├── app.module.ts
+        │   └── app-routing.module.ts
+        │
+        └── features                 # A feature is a functionality of the app. Contains Domain + Data layer.
+           └── feature-name          # Each feature contains his own domain and data logic
+                ├── data
+                │   ├── data-source
+                │   ├── entity
+                │   ├── mapper
+                │   └── query
+                ├── domain
+                │   ├── interactor
+                │   └── model
```

Comment:
Shouldn't these be in plural?

```suggestion
                ├── data
                │   ├── data-sources
                │   ├── entities
                │   ├── mappers
                │   └── queries
                ├── domain
                │   ├── interactors
                │   └── models
```

Reply:
Not as we have now. Let's decide it quickly because I'd like to have this merged. We can add more angular specific folders if we need.

In the current reference all this folders are in singular, using the name of the class that they contain. Other folders like "screens" or "features" are in plural because it's a more abstract grouping of classes.

I think it's a bit confusing since we might end up in subjective decisions about naming. Shall we use plural eveywhere? Is this ok for everybody? @lucianosantana @doup @Alex-DA @joselufo @franmontiel ?

This agreement should be for all the platforms.

Reply:
Managing namings in all the platforms and projects for plurals will be complex (we shouldn't be super strict with this), but I'm happy to keep it as plural.

Reply:
Naming updated. Resolving conversation and merging